pkg/cli/build.go

```diff
@@ -289,14 +289,19 @@ func GenBuildOptions(c *cobra.Command, inputArgs []string, iopts BuildOptions) (
 		timestamp = &t
 	}
 	if c.Flag("output").Changed {
-		buildOption, err := parse.GetBuildOutput(iopts.BuildOutput)
+		buildOption, err := parse.GetBuildOutput(iopts.BuildOutput, output)
 		if err != nil {
 			return options, nil, nil, err
 		}
 		if buildOption.IsStdout {
 			iopts.Quiet = true
 		}
 	}
+	if c.Flag("push").Changed {
+		if len(iopts.BuildOutput) == 0 {
```

Comment:
If both of the options are specified, that should fail instead of silently ignoring one.